apps/sports-helsinki/src/domain/search/searchHeader/SearchHeader.tsx

```diff
@@ -25,77 +32,94 @@ type Props = {
 
 function SearchHeader({ showMode, count, switchShowMode, searchForm }: Props) {
   const { t } = useCommonTranslation();
-  const [collapsed, setCollapsed] = useState<boolean>(true);
+  const [collapsed, setCollapsed] = useState<boolean>(false);
   const formWrapper = useRef<HTMLDivElement>(null);
 
   return (
     <>
       {showMode === ShowMode.MAP && (
         <div ref={formWrapper} className={styles.searchHeader}>
-          {!collapsed && (
-            <div className={styles.searchMenu}>
-              {searchForm}
-              <div className={styles.closeSearchWrapper}>
-                <Button
-                  className={styles.closeSearch}
-                  variant="secondary"
-                  theme="black"
-                  iconLeft={<IconCross aria-hidden="true" />}
-                  fullWidth
-                  onClick={() => setCollapsed(true)}
+          <>
+            {collapsed && (
+              <div>
+                <PageSection
+                  korosBottom={true}
+                  korosBottomClassName={styles.searchContainerKoros}
+                  className={classNames(styles.searchContainer, {})}
                 >
-                  {t('common:mapSearch.closeSearchControls')}
-                </Button>
+                  <ContentContainer
+                    className={classNames(styles.contentContainer, {})}
+                  >
+                    {searchForm}
+                  </ContentContainer>
+                </PageSection>
               </div>
-            </div>
-          )}
-          {collapsed && (
-            <div className={styles.searchActions}>
-              <Button
-                variant="secondary"
-                theme="black"
-                iconLeft={<IconMenuHamburger aria-hidden="true" />}
-                onClick={switchShowMode}
-              >
-                {t('common:mapSearch.showAsList')}
-              </Button>
-              <div className={styles.countAndTags}>
-                <Text variant="h3">
-                  {count} {t('common:mapSearch.searchResultsCountLabel')}
-                </Text>
-              </div>
-              <Button
-                variant="secondary"
-                theme="black"
-                iconLeft={<IconSearch aria-hidden="true" />}
-                onClick={() => {
-                  setCollapsed(false);
+            )}
+            {!collapsed && (
+              <PageSection>
+                <div className={styles.horizontalDivider}></div>
+              </PageSection>
+            )}
```

Comment:
An old code refactoring advice: I think it would make sense to move this part to a new (private) function / component or even split it in 2 new functions. It would make the code cleaner.